ledger/participant-integration-api/src/main/scala/platform/apiserver/RateLimitingInterceptor.scala

```diff
@@ -15,49 +16,83 @@ import org.slf4j.LoggerFactory
 private[apiserver] final class RateLimitingInterceptor(metrics: Metrics, config: RateLimitingConfig)
     extends ServerInterceptor {
 
-  import metrics.daml.lapi.threadpool.apiServices
-
   private val logger = LoggerFactory.getLogger(getClass)
 
   /** Match naming in [[com.codahale.metrics.InstrumentedExecutorService]] */
-  private val submitted = metrics.registry.meter(MetricRegistry.name(apiServices, "submitted"))
-  private val running = metrics.registry.counter(MetricRegistry.name(apiServices, "running"))
-  private val completed = metrics.registry.meter(MetricRegistry.name(apiServices, "completed"))
+  private case class InstrumentedCount(name: String, prefix: MetricName) {
+    private val submitted = metrics.registry.meter(MetricRegistry.name(prefix, "submitted"))
+    private val running = metrics.registry.counter(MetricRegistry.name(prefix, "running"))
+    private val completed = metrics.registry.meter(MetricRegistry.name(prefix, "completed"))
+    def queueSize: Long = submitted.getCount - running.getCount - completed.getCount
+  }
+
+  private val apiServices =
+    InstrumentedCount("Api Services", metrics.daml.lapi.threadpool.apiServices)
+  private val indexDbThreadpool = InstrumentedCount(
+    "Index Database Connection Threadpool",
+    MetricName(metrics.daml.index.db.threadpool.connection, ServerRole.ApiServer.threadPoolSuffix),
+  )
 
   override def interceptCall[ReqT, RespT](
       call: ServerCall[ReqT, RespT],
       headers: Metadata,
       next: ServerCallHandler[ReqT, RespT],
   ): ServerCall.Listener[ReqT] = {
 
-    val queued = submitted.getCount - running.getCount - completed.getCount
+    serviceOverloaded(call.getMethodDescriptor.getFullMethodName) match {
+      case Some(errorMessage) =>
+        val rpcStatus = com.google.rpc.Status
```

Comment:
This error is generated outside of the self-service error code framework. 
It needs to be converted to be compliant, it seems to me `ParticipantBackpressure.Rejection` fits the bill. In this way we will get the same automatic documentation and categorization that is already shared by all the other errors.

Reply:
Self service error codes are due to be done as part of https://digitalasset.atlassian.net/browse/DPP-1000